pkg/asset/installconfig/powervs/session.go

```diff
@@ -273,6 +278,153 @@ func (c *BxClient) ValidateCloudConnectionInPowerVSRegion(ctx context.Context, s
+	for i, m := range controlPlanes {
+		ctrplConfigs[i], ok = m.Spec.ProviderSpec.Value.Object.(*machinev1.PowerVSMachineProviderConfig)
+		if !ok {
+			return errors.New("m.Spec.ProviderSpec.Value.Object failed")
+		}
+	}
+	computeSystemType = ctrplConfigs[0].SystemType
+	computeProcessorType = string(ctrplConfigs[0].ProcessorType)
+	if ctrplConfigs[0].Processors.Type == intstr.Int {
+		computeProcessors = float64(numCompute) * float64(ctrplConfigs[0].Processors.IntVal)
+	} else {
+		cores, err := strconv.ParseFloat(ctrplConfigs[0].Processors.StrVal, 64)
+		if err != nil {
+			return errors.Wrap(err, "failed to convert compute cores to a float")
+		}
+		computeProcessors = float64(numCompute) * cores
+	}
+	computeMemoryGiB = int64(numCompute) * int64(ctrplConfigs[0].MemoryGiB)
+
+	// Find out the worker information
+	computeReplicas := make([]int64, len(computes))
+	computeConfigs := make([]*machinev1.PowerVSMachineProviderConfig, len(computes))
+	for i, w := range computes {
+		computeReplicas[i] = int64(*w.Spec.Replicas)
+		numWorker = computeReplicas[i]
+		computeConfigs[i], ok = w.Spec.Template.Spec.ProviderSpec.Value.Object.(*machinev1.PowerVSMachineProviderConfig)
+		if !ok {
+			return errors.New("w.Spec.Template.Spec.ProviderSpec.Value.Object")
+		}
+
+		workerSystemType = computeConfigs[i].SystemType
+		workerProcessorType = string(computeConfigs[i].ProcessorType)
+		if computeConfigs[i].Processors.Type == intstr.Int {
+			workerProcessors = float64(computeReplicas[i]) * float64(computeConfigs[0].Processors.IntVal)
+		} else {
+			cores, err := strconv.ParseFloat(computeConfigs[0].Processors.StrVal, 64)
+			if err != nil {
+				return errors.Wrap(err, "failed to convert worker cores to a float")
+			}
+			workerProcessors = float64(computeReplicas[i]) * cores
+		}
+		workerMemoryGiB += numWorker * int64(computeConfigs[i].MemoryGiB)
+	}
+
+	// Helpful debug statement to save typing
+	// fmt.Printf("ValidateCapacityWithPools: compute(%v) = {%v, %v, %v, %v}, worker(%v) = {%v, %v, %v, %v}\n", numCompute, computeSystemType, computeProcessorType, computeProcessors, computeMemoryGiB, numWorker, workerSystemType, workerProcessorType, workerProcessors, workerMemoryGiB)
+
+	switch computeProcessorType {
+	case "Dedicated":
+	case "Shared":
+		// @TODO I would think we should reduce the number of cores by some factor.
+		// However, I cannot currently find documentation which describes what
+		// PowerVS uses internally.
+	default:
+		return errors.Errorf("Unknown compute processor type (%v)", computeProcessorType)
+	}
+
+	switch workerProcessorType {
+	case "Dedicated":
+	case "Shared":
+		// @TODO I would think we should reduce the number of cores by some factor.
+		// However, I cannot currently find documentation which describes what
+		// PowerVS uses internally.
+	default:
+		return errors.Errorf("Unknown worker processor type (%v)", workerProcessorType)
+	}
+
+	for _, systemPool := range systemPools {
```

Comment:
I think we should not check cores at all in the Shared case for now. It's the most common case and we shouldn't overestimate how many cores we're using.

Reply:
Setting the number of CPUs to 0 in this case essentially disables the check,